manifests/tasks_service.pp

```diff
@@ -0,0 +1,20 @@
+# = foreman-tasks service
+#
+# Service for the foreman-tasks service, which starts a Dynflow executor
+# for background job processing in Foreman. Not to be confused with the
+# foreman-tasks plugin.
+#
+# === Parameters:
+#
+# $service:: Service name
+#            type:String
+
+class foreman::tasks_service(
+  $service = $::foreman::params::tasks_service,
+) inherits foreman::params {
+  service { 'foreman-tasks':
+    ensure => running,
```

Comment:
I'd suggest pulling the variables out so you can overwrite them if you want.
Just move them to the "signature" of the class, similar to service.